Classes/Security/HashToken.php

```diff
@@ -16,22 +17,25 @@ class HashToken extends AbstractToken
 
     /**
      * @param ActionRequest $actionRequest
-     * @return bool|void
-     * @throws \Neos\Flow\Security\Exception\InvalidAuthenticationStatusException
+     * @return bool
+     * @throws InvalidAuthenticationStatusException
      */
-    public function updateCredentials(ActionRequest $actionRequest)
+    public function updateCredentials(ActionRequest $actionRequest): bool
```

Comment:
The return value is not actually used (see https://github.com/neos/flow-development-collection/blob/af7b3374688878b822528b4a761741f1102de1cf/Neos.Flow/Classes/Security/Context.php#L787) and IMO it should be `void` in the interface (it already _is_ in the Flow implementations).

We should at least avoid declaring a return type hint to avoid this part from breaking if it gets fixed in the interface:
```suggestion
    public function updateCredentials(ActionRequest $actionRequest)
```